pkg/provision/storage/cleanup.go

```diff
@@ -203,3 +226,43 @@ func commonPVCExists(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) (bo
 	}
 	return true, nil
 }
+
+func checkCleanupPodsState(job *batchv1.Job, workspaceID string, clusterAPI sync.ClusterAPI) (msg string, err error) {
+	pods := &corev1.PodList{}
+	if err := clusterAPI.Client.List(context.TODO(), pods, k8sclient.InNamespace(job.Namespace), k8sclient.MatchingLabels{"job-name": common.PVCCleanupJobName(workspaceID)}); err != nil {
+		return "", err
+	}
+
+	for _, pod := range pods.Items {
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if status.CheckContainerStatusForFailure(&containerStatus) {
```

Comment:
I knew something was wrong... this should be negated, ie. `!status.CheckContainerStatusForFailure(&containerStatus)`